python/lsst/sphgeom/__init__.py

```diff
@@ -37,3 +38,130 @@
 from .version import *
 
 PixelizationABC.register(Pixelization)
+
+# Copy and paste from lsst.utils.wrappers
+_INTRINSIC_SPECIAL_ATTRIBUTES = frozenset(
+    (
+        "__qualname__",
+        "__module__",
+        "__metaclass__",
+        "__dict__",
+        "__weakref__",
+        "__class__",
+        "__subclasshook__",
+        "__name__",
+        "__doc__",
+    )
+)
+
+
+def _isAttributeSafeToTransfer(name: str, value: typing.Any) -> bool:
+    if name.startswith("__") and (
+        value is getattr(object, name, None) or name in _INTRINSIC_SPECIAL_ATTRIBUTES
+    ):
+        return False
+    return True
+
+
+def _continueClass(cls):
+    import sys
+
+    orig = getattr(sys.modules[cls.__module__], cls.__name__)
+    for name in dir(cls):
+        # Common descriptors like classmethod and staticmethod can only be
+        # accessed without invoking their magic if we use __dict__; if we use
+        # getattr on those we'll get e.g. a bound method instance on the dummy
+        # class rather than a classmethod instance we can put on the target
+        # class.
+        attr = cls.__dict__.get(name, None) or getattr(cls, name)
+        if _isAttributeSafeToTransfer(name, attr):
+            setattr(orig, name, attr)
+    return orig
+
+
+def _inf_to_lat(lat: float) -> float:
+    """Map +Inf to +90 and -Inf to -90 degrees."""
+    import math
```

Comment:
Is this to keep `math` from module-level `vars`? I'm worried that there is non-zero price to pay for this on each call.

Reply:
Ok. I will see about moving the entire content into its own file so that there won't be leakage into the caller namespace.

Reply:
You can add `del math` at the bottom like you do with `typing`.

Reply:
Yes, although for reasons unexplained I couldn't get math to work yesterday. I'm seeing if I can move it into a separate file.